pkg/kritis/attestation/attestation.go

```diff
@@ -53,27 +52,40 @@ func VerifyMessageAttestation(pubKeyEnc string, attestationHash string, message
 	if err != nil {
 		return err
 	}
+
 	attestation, err := base64.StdEncoding.DecodeString(attestationHash)
 	if err != nil {
 		return err
 	}
 
-	keyring, err := openpgp.ReadArmoredKeyRing(strings.NewReader(string(pemPublicKey)))
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(pemPublicKey))
 	if err != nil {
 		return err
 	}
 	buf := bytes.NewBuffer([]byte(attestation))
 	armorBlock, err := armor.Decode(buf)
+	if err != nil {
+		return errors.Wrap(err, "could not decode armor signature")
+	}
 	md, err := openpgp.ReadMessage(armorBlock.Body, keyring, nil, &pgpConfig)
+	if err != nil {
+		return errors.Wrap(err, "could not read armor signature")
+	}
 
-	// Verify Signature using the Public Key
+	// MessageDetails.UnverifiedBody signature is not verified until we read it.
+	// This will call PublicKey.VerifySignature for the keys in the keyring.
 	plaintext, err := ioutil.ReadAll(md.UnverifiedBody)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "could not verify armor signature")
+	}
+	// Make sure after reading the UnverifiedBody above, there is no signature error.
+	if md.SignatureError != nil || md.Signature == nil {
+		return fmt.Errorf("Bad signature found: %s or No signature found for given key", md.SignatureError)
 	}
 
+	// Finally, make sure the signature is over the right message.
 	if string(plaintext) != message {
-		return fmt.Errorf("Signature could not be verified. Got: %q, Want: %q", plaintext, message)
+		return fmt.Errorf("Signature could not be verified. Got: %s, Want: %s", string(plaintext), message)
```

Comment:
https://github.com/golang/go/wiki/Errors

Reply:
thanks! will fix!

Reply:
Also, why call string(plaintext) - won't fmt.Errorf do that for you?

Reply:
yes it does. copy paste issue.